src/app/components/msel-list/msel-list.component.html

```diff
@@ -133,7 +49,49 @@
       </button>
     </div>
   </div>
-
+<ng-container>
+  <div class="buttons-container">
+    <div class="button-items">
+      <button mat-icon-button (click)="addMsel(); $event.stopPropagation()" title="Add blank MSEL" [disabled]="!isContentDeveloper">
+        <mat-icon class="mdi-24px" fontIcon="mdi-plus-circle-outline"></mat-icon> Blank
+      </button>
+    </div>
+    <div class="button-items">
+      <button mat-icon-button [matMenuTriggerFor]="upload" title="Upload a new MSEL from a file" [disabled]="!isContentDeveloper">
+        <mat-icon class="mdi-24px" fontIcon="mdi-upload"></mat-icon> Upload
+      </button>
+      <mat-menu #upload="matMenu" [overlapTrigger]="false">
+        <button mat-menu-item (click)="uploadFile('json', '', ''); jsonInput.click()" title="Upload json file">Upload json file</button>
+        <button mat-menu-item (click)="uploadFile('xlsx', '', ''); xlsxInput.click()" title="Upload xlsx file">Upload xlsx file</button>
+      </mat-menu>
+    </div>
+    <div class="button-items">
+      <mat-select [(ngModel)]="selectedMselType" (selectionChange)="filterMsels()">
+        <mat-option value="all">All Types</mat-option>
+        <mat-option value="is">Templates</mat-option>
+        <mat-option value="is not">Not Templates</mat-option>
+      </mat-select>
+    </div>
+    <div class="button-items">
+      <mat-select [(ngModel)]="selectedMselStatus" (selectionChange)="filterMsels()">
+        <mat-option value="all">All Statuses</mat-option>
+        <mat-option value="Pending">Pending</mat-option>
+        <mat-option value="Entered">Entered</mat-option>
+        <mat-option value="Approved">Approved</mat-option>
+        <mat-option value="Completed">Completed</mat-option>
+      </mat-select>
+    </div>
+    <div class="button-items search-box">
+      <mat-icon class="search-icon" svgIcon="ic_magnify_search"></mat-icon>
+      <mat-form-field class="search-control">
+        <input matInput [formControl]="filterControl" placeholder="Search" />
+      </mat-form-field>
+      <button mat-icon-button (click)="filterControl.setValue(''); $event.stopPropagation()" title="Clear Search" [disabled]="!filterString">

```

Comment:
Same thing with this $event.stopPropagation().

Reply:
Same here.